libc-top-half/musl/src/thread/wasm32/wasi_thread_start.s

```diff
@@ -0,0 +1,31 @@
+	.text
+
+	.export_name	wasi_thread_start, wasi_thread_start
+
+	.globaltype	__stack_pointer, i32
+	.globaltype	__tls_base, i32
+	.functype	__wasi_thread_start_C (i32, i32) -> ()
+
+	.hidden	wasi_thread_start
+	.globl	wasi_thread_start
+	.type	wasi_thread_start,@function
+
+wasi_thread_start:
+	.functype	wasi_thread_start (i32, i32) -> ()
+
+	# Set up the minimum C environment.
+	# Note: offsetof(start_arg, stack) == 0
+	local.get   1  # start_arg
+	i32.load    0  # stack
+	global.set  __stack_pointer
+
+	local.get   1  # start_arg
+	i32.load    4  # tls_base
+	global.set  __tls_base
```

Comment:
Would it be simpler just to pass these as two extra args to __wasi_thread_start_C?  This would avoid using memory at all, be smaller in terms of code size, and be less brittle in terms of struct offsets.

Reply:
> Would it be simpler just to pass these as two extra args to __wasi_thread_start_C? This would avoid using memory at all, be smaller in terms of code size, and be less brittle in terms of struct offsets.

i'm not sure what you mean.
do you mean to save a pthread_self() call in the C function?